core/modulesource.go

```diff
@@ -572,7 +572,7 @@ func (src *ModuleSource) ModuleConfig(ctx context.Context) (*modules.ModuleConfi
 	configFile, err := contextDir.Self.File(ctx, filepath.Join(rootSubpath, modules.Filename))
 	if err != nil {
 		// no configuration for this module yet, so no name
-		return nil, false, nil
+		return nil, false, nil //nolint:nilerr
```

Comment:
Is this case not intentional? The `nolint` inline comment, and the comment above feels deliberate.

I feel like this might be possible when initializing a module for the first time in an empty directory maybe?

Reply:
Not following, the behavior is intentional to return nil err. I added the nolint comment so that the linter didn't yell at us for this case